src/EventStore.Core/Services/Storage/ReaderIndex/AllReader.cs

```diff
@@ -48,7 +48,7 @@ public class AllReader : IAllReader {
 		}
 
 		public IndexReadAllResult ReadAllEventsForward(TFPos pos, int maxCount) {
-			return ReadAllEventsForwardInternal(pos, maxCount, maxCount, EventFilter.None);
+			return ReadAllEventsForwardInternal(pos, maxCount, int.MaxValue, EventFilter.DefaultAllFilter);
```

Comment:
hmmm, i think that the reason for setting the `maxSearchWindow` to `maxCount` is to add an upper bound to the read, although a multiple of maxCount (e.g 3 x maxCount) sounds more reasonable.